src/transformers/models/clvp/modeling_clvp.py

```diff
@@ -1870,6 +1874,13 @@ def generate(
         generation_config.validate()
         self._validate_model_kwargs(model_kwargs.copy())
 
+        # pad input_ids as specified in the original repo
+        # link: https://github.com/neonbjb/tortoise-tts/blob/80f89987a5abda5e2b082618cd74f9c7411141dc/tortoise/api.py#L380
+        input_ids = torch.nn.functional.pad(input_ids, (0, 1))
+        attention_mask = (
+            torch.nn.functional.pad(attention_mask, pad=(0, 1), value=1) if attention_mask is not None else None
+        )
+
```

Comment:
One question: Is it ok to keep it here? Or should I move this to the tokenizer file since it's closely related to the tokenizer?

Reply:
I would keep this here: this is a modelling specific modification which always pads and only when generating. For the tokenizers, we expect them to behave in a standard way i.e. `padding=False` will not pad the inputs and `padding=True` will pad to the max sequence length.  cc @ArthurZucker WDYT?

Two general comments: 
* For the attention masks, 0 indicates tokens to ignore, and 1 those to attend. If sequences are padded, then we can end up with unusual attentions masks like: `[1, 1, 1, 0, 0, 0, 0, 0, 1]`. Effectively we will always attend to the last token `""`. I'm surprised this is done - regardless of the length of the input sentence we're always adding a token which will have a positional encoding of the largest possible sequence length. Do we know why the original researchers did this? 
*  Won't this break other assumptions in the model? At the moment doing a forward pass, we can pass in either a batch of sequences, all padded to the max sequence length in that batch. or all padded to the max possible sequence length of the model. If the latter is passed, this would make the input_ids 1 token longer than they can be.

Reply:
> Do we know why the original researchers did this? 

No, 

> If sequences are padded, then we can end up with unusual attentions masks like: [1, 1, 1, 0, 0, 0, 0, 0, 1]. Effectively we will always attend to the last token ""

In the official repo, they don't process texts in batches so no `attention_mask` and no issue there :) .

> this would make the input_ids 1 token longer than they can be.

To remove this possibility, instead of adding it here I think this line should be added to the pipeline code in the diffusers library where we check for the length of the sequence before passing it to the model. WDYT?

Reply:
> No,

OK - I suggest raising an issue on the repo to ask

> In the official repo, they don't process texts in batches so no attention_mask and no issue there :) .

What we'd need to protect against is people using batches in our repo - how would a user know not to do this? We'd have to do something like raising an exception if they try

> To remove this possibility, instead of adding it here I think this line should be added to the pipeline code in the diffusers library where we check for the length of the sequence before passing it to the model. WDYT?

It really depends on what this code is for (which we need to find out). All modeling logic should be contained within the model itself. I'd expect the sequence length info to be accessible in `self.config`

Reply:
Hi @amyeroberts, I talked with @manmay-nakhashi (the maintainer of the `tortoise-tts` repo) on Slack and he confirmed that instead of adding the `0` at the end, it should be added where the valid `input_ids` end, so that the `0` will be considered as a valid token and we will have usual `attention_mask`. So,

input_ids - `[5, 6, 7, [PAD], [PAD]]`, attention_mask - `[1, 1, 1, 0, 0]`

After adding the `0` - 

input_ids - `[5, 6, 7, 0, [PAD], [PAD]]`, attention_mask - `[1, 1, 1, 1, 0, 0]`

> this would make the input_ids 1 token longer than they can be.

To avoid this I think we should just check for the `self.config.decoder_config.max_text_tokens - 3` and raise an error if the user gives longer tokens than that. The 3 is because we already add bos and eos tokens for the [ClvpConditioningEncoder](https://github.com/huggingface/transformers/blob/f18c95b49cbb675764f0cd19b56fb70f39a62bb2/src/transformers/models/clvp/modeling_clvp.py#L604) and we are adding one more extra token here :)   

WDYT?

Reply:
@susnato Thanks for looking into this and writing this details explanation! 

In principle sounds good to me. Could you link to where you think the exception should be raised?